### DISCUSSION TITLE
Is it possible to rename @@unique constraint?

### DISCUSSION THREAD

**Discussion by michael-land** [prisma/prisma#7488] [Q&A] [answered]
```sql
-- auto-generated definition
CREATE TABLE company
(
  id                       INTEGER GENERATED BY DEFAULT AS IDENTITY
    CONSTRAINT company_pk
      PRIMARY KEY,
  ...other fields

  CONSTRAINT company_channel_id_external_id_uk
    UNIQUE (channel_id, external_id)
);
```

```prisma
model Company {
  id                    Int                      @id @default(autoincrement())
  channelId             Int                      @map("channel_id") @db.SmallInt
  externalId            String                   @map("external_id")
 
  @@unique([channelId, externalId], name: "company_channel_id_external_id_uk")
  @@map("company")
}
```

```ts
await prisma.company.upsert({
    where: { company_channel_id_external_id_uk: { channelId, externalId } },
    create: data,
    update: data,
  });
```

I want to keep the constraint name snakecase in database `company_channel_id_external_id_uk`, but want to rename it to `companyChannelIdExternalId` in prisma. is it possible?

**Comment by ryands17** [comment, marked as answer]
@xiaoyu-tamu 👋 

I don't think this is possible. I would suggest opening a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with your use case so that we can look into this.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
